### ISSUE TITLE
Curseforge API impacts

### ISSUE THREAD

**Issue by Madis0** [Fabulously-Optimized/fabulously-optimized#169]
A summarizing issue to list the potential impacts for this modpack by [the upcoming Curseforge API](https://mailchi.mp/overwolf/whats-new-with-overwolf-curseforge-november3).

Notable points:

- ~~As a default, the toggle [to allow accessing a CF project with the API] will be on for existing projects, and off for new ones.~~
Edit: I've heard that all new projects _must now explicitly select_ whether they do or do not want to allow external downloads. Still, same caveats apply if disallowed.
  - Translation: new mods added to the pack may not be downloadable by non-CurseForge launchers, that could cause fragmentation
  - Solution: FO will keep it enabled and suggest mod authors to do the same
-  Downloading a modpack through a 3rd party service using the API will download only the enabled mods [or dependency mods]. 
    - Translation: modpack may lack some mods when downloaded via a non-Curseforge Launcher
    - Solutions:
        - For MultiMC: make sure to download from FO's CurseForge listing, not from its built-in feature
        - Download the missing mods manually
        - Switch to CurseForge, MultiMC or vanilla launcher
- Legacy API is deprecated on ~~February~~ 9th of May
  - Translation: outdated non-CurseForge launchers will break, but I expect the ones I recommend to update in time
  - Solution: use Curseforge Launcher, MultiMC or vanilla launcher if the others do break
- ~~Curseforge linux support _should_ come out before the old API is killed off. [source: CurseForge discord moderator]~~ Edit: not exactly, but it is still coming soon
  - Translation: Linux users can also use CurseForge Launcher, which means better experience for them, supporting me and the mods
  - [Also do vote for this survey!](https://forms.gle/9qmiJDbZV79RJJbS9)

**Comment by Madis0** [comment]
* CF API postponed to April
* Linux CF client soon, but WoW only at first
* You can now install individual mods to a CF profile via the install button on the site (presumably you still need to enable content management toggle first, haven't tested)
https://mailchi.mp/overwolf/whats-new-with-overwolf-curseforge-february22

**Comment by Madis0** [comment]
> hopefully, the end of this month [there is info regarding the migration]

- wolfgang, a Overwolf dev [(via)](https://discord.gg/overwolf-developers)

**Comment by DioEgizio** [comment]
Hmm we are in May but the old API is still working

**Comment by Madis0** [comment]
> 9th of May
- wolfgang

**Comment by Madis0** [comment]
* API deprecated on 9th of May
* New website planned, with better navigation and search. Will be a separate beta portal at first, then merged into main.
* Linux client will be out for WoW in May, Minecraft and others to follow later.

https://us15.campaign-archive.com/?u=5542885c5f38ae9a4108ab154&id=feacea0385

**Comment by Madis0** [comment]
* CF postponed it to May 16th
* Linux client out for WoW

https://medium.com/overwolf/delay-in-the-old-api-deprecation-and-linux-client-launch-f424cbc4235c

* packwiz-installer will be updated in due course to support these changes, though I cannot guarantee this will be finished before the old API is disabled.
* Existing modpacks, with the download URLs already stored, will not be affected unless you wish to update or install new mods from CurseForge.

https://discord.com/channels/754677731702341672/782268590182498315/972537733157376051 (via https://discord.gg/DcSkRF4)

So basically current MMC (au) and Modrinth won't break and the dev is working to update the former to CF API. Modrinth will not be able to get updated via CF soon though, might figure out alternatives.

**Comment by Madis0** [comment]
https://medium.com/overwolf/the-curseforge-official-api-is-now-live-d314606355c2
TL;DR:
* New API released
* There is a Overwolf-free alpha version of CF Launcher for Windows, currently WoW-only

**Comment by Madis0** [comment]
Current status:

* packwiz is not yet updated, though working on it
* GDLauncher kills your entire instances when you update lmao (https://github.com/gorilla-devs/GDLauncher/issues/1288)
* MultiMC's Curseforge support is 🦀 (https://github.com/MultiMC/Launcher/commit/0a827ba70e6ef20187f8507a536d54a8441020dc)
* Modrinth's Curseforge support is iffy at best

Good times. /s

**Comment by GhostIsBeHere** [comment]
ATLauncher currently has a workaround to this, they will open the mod download in your browser, and then automatically  move it from your downloads folder to your mods folder (still a little buggy)
This works about 95% of the time

**Comment by Madis0** [comment]
Update summary:

* packwiz is fixed a long time ago
* Modrinth no longer supports CF URLs #63 
* GDLauncher finally fixed the downloads
* MultiMC doesn't intend to readd CF any time soon (https://github.com/MultiMC/Launcher/issues/4762#issuecomment-1152912827) though I don't really care for it anyway, haha
* PolyMC, however, has decent CF support

**Comment by DioEgizio** [comment]
> Overall currently only one opted out mod is Entity Culling

Wanted to add that it's available on modrinth though, so it's not a big issue for most third party launchers

**Comment by Madis0** [comment]
A month later, still only Entity Culling is opted out and no other timely issues exist anymore. Packwiz downloads it from Modrinth and GDLauncher downloads it via the embedded browser, so I'd say I can finally close this issue.
